### PR TITLE
Improvments to "Displayed Columns"

### LINKED ISSUE #355: Controller mode Dashboard - Displayed columns should default to selected
### Your Vatsim CID

1479381

### What do you suggest?

On the dashboard, in controller mode, on the Displayed Columns selector, all options should default to on, since all columns are showing up once you open the page.

### Why do you need that?

It makes it way easier to disable one or two columns.

Right now if you want to disable departures for example, you must open the selector, and then re-select all the columns, except departures.

If all of them defaulted to selected it would only require two clicks, on the selector, and on the column to disable.

### Alternatives

_No response_

### Additional info

I cloned the repo and am working on a fix. I believe the code is ready, I just want to test it thoroughly and add some comments. I'll make a pull request when everything is ready (1-2 days).
Yes, I am the same person that messaged you on discord, and I'm just adding this here so people know the feature is already "requested"

### Validation

- [X] I have searched for other requests and didn't found any alike
- [X] I have also checked roadmap, and this is not planned

### PR DESCRIPTION
**Pull request by Samuel-Fortunato**
### ❓ Type of change

- [ ] 🐞 Bug fix
- [X] 👌 Enhancement (improve something existing)
- [ ] ✨ New functionality
- [ ] 🧹 Technical change (refactoring, updates and other improvements)

### 📚 Description

This pull request changes the "Displayed Columns" selector in Controller Mode, so that all the columns come pre-selected, which makes it easier and more intuitive to disable one or more columns. The selected items will also be kept from session to session in a cookie.
Finally, I added a new Boolean prop to CommonSelector.vue, named showSelectedValue, which indicates if the Selector should display the selected item's name (or the number of selected items), or the Selector's placeholder name. It defaults to true, so all selectors behave in the same way, except for the Displayed Columns one which always shows the placeholder name. I thought this is better since there is no advantage in showing "# selected" when the columns are right below.
If you have a better idea for this final part feel free to change anything or tell me.

### CODE REVIEW

File: src/components/common/basic/CommonSelect.vue [hunk 1]
```diff
@@ -71,6 +71,17 @@ const props = defineProps({
...
         type: String,
     },
+    /** showSelectedValue
+      *
+      * Switches between showing
+      * the selected items or the
+      * placeholder name.
+      * @default true
+      */
+    showSelectedValue: {
```
Comment (daniluk4000) [inline src/components/common/basic/CommonSelect.vue:81]: Let's rename this to showPlaceholder with false by default

File: src/pages/airport/[icao].vue [hunk 2]
```diff
@@ -351,7 +352,12 @@ const aircraftModes: SelectItem<MapAircraftMode>[] = [
     },
 ];
 
-const displayedColumns = ref<MapAircraftKeys[]>([]);
+const displayedColumns = useCookie<MapAircraftKeys[]>('dashboard-displayed-columns', {
```
Comment (daniluk4000) [inline src/pages/airport/[icao].vue:355]: Please remove useCookie and go back to array mode. We'll have presets for this in 0.5.0 hopefully

**Review by daniluk4000 — CHANGES_REQUESTED** [review]

File: src/pages/airport/[icao].vue (hunk 2 above)
Comment (Samuel-Fortunato) [reply src/pages/airport/[icao].vue:355]: Do I keep the selected by default feature, but with array mode?

Comment (daniluk4000) [reply src/pages/airport/[icao].vue:355]: Sure! Just update array with same keys as you did here in default

**Comment by daniluk4000** [comment]
Thanks!

**Pull request merged**